xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/PoloniexStreamingMarketDataService.java

```diff
@@ -90,12 +79,26 @@ public PoloniexStreamingMarketDataService(PoloniexStreamingService service, Map<
                 .share();
 
         return subscribedTrades
-                .map(s -> adaptPoloniexPublicTrade(s.toPoloniexPublicTrade(currencyPair), currencyPair));
+                .map(s -> adaptPoloniexPublicTrade(s, currencyPair));
     }
 
-    private Optional<PoloniexOrderbook> getPoloniexOrderbook(
-            final Optional<PoloniexOrderbook> orderbook, final PoloniexWebSocketEvent s
-    ) {
+    private static Trade adaptPoloniexPublicTrade(
+            PoloniexWebSocketTradeEvent poloniexTradeEvent, CurrencyPair currencyPair) {
+        TradeEvent tradeEvent = poloniexTradeEvent.getTradeEvent();
+        Date timestamp = new Date(tradeEvent.getTimestampSeconds() * 1000);
+        Trade trade =
+                new Trade(
```

Comment:
Better to use `Trade.Builder`

Reply:
why? I think builder is redundant here because all fields are always required... so you don't have any flexibility in building, thus don't need Builder

Reply:
Because then the code doesn't get broken when new fields are added to `Trade`, as has happened several times.

Also,IIRC walec intends to remove the publicly visible constructors in XChange in the future.

Reply:
maybe is should get broken for example if new fields are required?:) I suggest we create a ticket to make an improvement in all places instead of just this one and implement it later. What do you think?

Reply:
I'm happy with just progressive improvement, but if you felt like doing a widespread cleanup PR that would obviously be really appreciated!

Reply:
fixed